src/tls.c

```diff
@@ -4866,21 +4864,21 @@ int TLSX_ValidateSupportedCurves(WOLFSSL* ssl, byte first, byte second) {
 #endif
     {
         key = 1;
-        ssl->ecdhCurveOID = defOid;
+        cs->ecdhCurveOID = defOid;
```

Comment:
ssl->ecdhCurveOID was being updated before.
Now we check ssl->ecdhCurveOID without updating.
Is this correct behaviour?

Reply:
Changing all checks to `*ecdhCurveOID` and setting initial value to `*ecdhCurveOID = ssl->ecdhCurveOID;`. This should replicate the previous behaviour.